### PR TITLE
of0.98->of~0.10.git

### PR DESCRIPTION
**Pull request by gllmAR**
changed I had to make in order to compile ofMidi with of~0.10.git
changed: 

* string to std::string, 
* vector to std::vector
* ofPtr to std::shared_ptr  
* included ofLog.h where logging happens

### CODE REVIEW

File: src/desktop/ofxRtMidiIn.h [hunk 1]
```diff
@@ -17,26 +17,29 @@ class ofxRtMidiIn : public ofxBaseMidiIn {
...
 	void ignoreTypes(bool midiSysex, bool midiTiming, bool midiSense);
 
 private:
 	
 	RtMidiIn midiIn;
-	static ofPtr<RtMidiIn> s_midiIn; //< for static calls (a little kludgey, I know)
-	
+//	static ofPtr<RtMidiIn> s_midiIn; //< for static calls (a little kludgey, I know)
+
+    static std::shared_ptr<RtMidiIn> s_midiIn; //< for static calls (a little kludgey, I know)
```
Comment (danomatika) [inline src/desktop/ofxRtMidiIn.h]: Can fix indent and remove commented line above.

File: src/desktop/ofxRtMidiOut.h [hunk 2]
```diff
@@ -18,23 +18,23 @@ class ofxRtMidiOut : public ofxBaseMidiOut {
...
+	bool openVirtualPort(std::string portName);
 	void closePort();
 	
 private:
 	
 	void sendMessage();
 
 	RtMidiOut midiOut;
-	static ofPtr<RtMidiOut> s_midiOut; //< for static calls (a little kludgey, I know)
+    static std::shared_ptr<RtMidiOut> s_midiOut; //< for static calls (a little kludgey, I know)
```
Comment (danomatika) [inline src/desktop/ofxRtMidiOut.h]: Fix indent.

File: src/ofxBaseMidi.h [hunk 3]
```diff
@@ -25,16 +25,16 @@ class ofxBaseMidiIn {
...
 	virtual bool openPort(unsigned int portNumber) = 0;
-	virtual bool openPort(string deviceName) = 0;
-	virtual bool openVirtualPort(string portName) = 0;
+	virtual bool openPort(std::string deviceName) = 0;
+	virtual bool openVirtualPort(std::string portName) = 0;
 	virtual void closePort() = 0;
 	
 	int getPort();
-	string getName();
+    std::string getName();
```
Comment (danomatika) [inline src/ofxBaseMidi.h]: Fix indent.

File: src/ofxBaseMidi.h [hunk 4]
```diff
@@ -49,12 +49,12 @@ class ofxBaseMidiIn {
 protected:
 
 	/// parses and sends received raw messages to listeners
-	void manageNewMessage(double deltatime, vector<unsigned char> *message);
+	void manageNewMessage(double deltatime, std::vector<unsigned char> *message);
 	
 	int portNum;     //< current port num, -1 if not connected
-	string portName; //< current port name, "" if not connected
+    std::string portName; //< current port name, "" if not connected
```
Comment (danomatika) [inline src/ofxBaseMidi.h]: Fix indentation in this whole file.

File: src/ofxMidiIn.h [hunk 5]
```diff
@@ -69,7 +69,7 @@ class ofxMidiIn {
 	/// note: this order may change when new devices are added/removed
 	///       from the system
 	///
-	static vector<string>& getPortList();
```
Comment (danomatika) [inline src/ofxMidiIn.h]: Fix indentation in this file as well.

**Comment by danomatika** [comment]
Thanks. This will not be merged until OF 0.10 is released as the master branch is kept to the current OF release.

Also, please fix the indentation which uses tabs.

**Comment by gllmAR** [comment]
understood, thanks you.

**Comment by danomatika** [comment]
Also, you don't need to make 1 commit for each indent fix. Please put them all into 1 commit.

**Comment by gllmAR** [comment]
sorry,  I think that my last commit introduced some changes in the tab width, i'll try to understand it out.

**Comment by hamoid** [comment]
I did all these changes today on my copy (had no Internet to find this PR). I also 
* replaced `NULL` with `nullptr` to avoid compiler warnings
* included `<cmath>` in `ofxMidi.cpp` (couldn't find `log` and `exp` for some reason)

**Comment by vooku** [comment]
Ping! I'm glad I found this PR, but not having to look for it would be even better as openFrameworks 0.10 has been out for a month.

**Comment by vooku** [comment]
I'm sorry, didn't realise I was being this rude.

**Comment by danomatika** [comment]
All: I will probably get to this in a week or two when I update all of my addons.

**Comment by danomatika** [comment]
Fixed with latest commits.

**Pull request closed without merging**